src/transform/free_vars.rs

```diff
@@ -144,7 +144,7 @@ fn collect_free_vars(rt: &mut RichTerm, free_vars: &mut HashSet<Ident>) {
             *deps = Some(new_deps);
         }
         Term::Array(ts, _) => {
-            for t in ts {
+            for t in ts.make_mut().iter_mut() {
```

Comment:
Nitpick, but same here: I would let user code inchanged (consumer of arrays), and add an `iter_mut()` wrapper inside `Array`, such that using them feel almost transparent, as if we didn't do any change. Future contributors also won't have the additional mental load of having to know those small discrepancies between using our `Array` and the standard `Vec`.

Reply:
Actually, I take that back: the difference is now that `iter_mut()` may clone the entire backing array, which is not the case with a usual vector. So it may actually be a good thing that the caller is aware of that difference.